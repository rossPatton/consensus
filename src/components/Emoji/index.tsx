import React, {memo} from 'react';

const Emoji = memo((props: {label: string, emoji: string}) => (
  <small
    role="img"
    aria-label={props.label}>
    {props.emoji}
  </small>
));

export default Emoji;
