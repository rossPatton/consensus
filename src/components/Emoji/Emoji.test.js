import React from 'react';
import render from 'react-test-renderer';

import Emoji from '.';

it('renders without crashing', () => {
  render.create(<Emoji />);
});

it('renders null when no props passed in', () => {
  const component = render.create(<Emoji />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders as expected', () => {
  const component = render.create((
    <Emoji
      label="testing butt emoji"
      emoji="🍑"
    />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
