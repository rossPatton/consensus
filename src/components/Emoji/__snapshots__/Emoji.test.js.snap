// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<small
  aria-label="testing butt emoji"
  role="img"
>
  🍑
</small>
`;

exports[`renders null when no props passed in 1`] = `null`;
